src/main/java/com/microsoft/jfr/generation/JfcParser.java

```diff
@@ -0,0 +1,97 @@
+package com.microsoft.jfr.generation;
+
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+class JfcParser {
+
+    private final Map<String, List<Setting>> settingsByEventName = new LinkedHashMap<>();
+    private final String sourceFilePath;
+    private final SAXParserFactory factory = SAXParserFactory.newInstance();
+
+    JfcParser(String sourceFilePath) {
+        this.sourceFilePath = sourceFilePath;
+    }
+
+    Map<String, List<Setting>> getSettingsByEventName() {
+        return settingsByEventName;
+    }
+    void parse() {
+        try {
+            SAXParser saxParser = factory.newSAXParser();
+            saxParser.parse(sourceFilePath, new JfcToMap(this));
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            throw new RuntimeException(e);
```

Comment:
Can you throw a checked exception here? Maybe JfrStreamingException would work. 

If this is going to be used in an agent, for example, the RuntimeException will bring down the whole app.

Reply:
The above comment may be because I misunderstand the use case. After reviewing the other files, I believe what you're doing here is providing a utility that transforms a jfc file to a Java class file, rather than something that could be used at runtime to parse a jfc file into some kind of domain-specific class.